packages/std/src/math/signed_decimal_256.rs

```diff
@@ -0,0 +1,3226 @@
+pub struct SignedDecimal256(#[schemars(with = "String")] Int256);
+
+forward_ref_partial_eq!(SignedDecimal256, SignedDecimal256);
+
+#[derive(Error, Debug, PartialEq, Eq)]
+#[error("SignedDecimal256 range exceeded")]
+pub struct SignedDecimal256RangeExceeded;
+
+impl SignedDecimal256 {
+    const DECIMAL_FRACTIONAL: Int256 = // 1*10**18
+        Int256::from_be_bytes([
+            0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 13, 224, 182,
+            179, 167, 100, 0, 0,
+        ]);
+    const DECIMAL_FRACTIONAL_SQUARED: Int256 = // 1*10**36
+        Int256::from_be_bytes([
+            0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 192, 151, 206, 123, 201, 7, 21, 179,
+            75, 159, 16, 0, 0, 0, 0,
+        ]);
+
+    /// The number of decimal places. Since decimal types are fixed-point rather than
+    /// floating-point, this is a constant.
+    pub const DECIMAL_PLACES: u32 = 18; // This needs to be an even number.
+
+    /// The largest value that can be represented by this signed decimal type.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// # use cosmwasm_std::SignedDecimal256;
+    /// assert_eq!(
+    ///     SignedDecimal256::MAX.to_string(),
+    ///     "57896044618658097711785492504343953926634992332820282019728.792003956564819967"
+    /// );
+    /// ```
+    pub const MAX: Self = Self(Int256::MAX);
+
+    /// The smallest value that can be represented by this signed decimal type.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// # use cosmwasm_std::SignedDecimal256;
+    /// assert_eq!(
+    ///     SignedDecimal256::MIN.to_string(),
+    ///     "-57896044618658097711785492504343953926634992332820282019728.792003956564819968"
+    /// );
+    /// ```
+    pub const MIN: Self = Self(Int256::MIN);
+
+    /// Creates a SignedDecimal256(value)
+    /// This is equivalent to `SignedDecimal256::from_atomics(value, 18)` but usable in a const context.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// # use cosmwasm_std::{SignedDecimal256, Int256};
+    /// assert_eq!(SignedDecimal256::new(Int256::one()).to_string(), "0.000000000000000001");
+    /// ```
+    pub const fn new(value: Int256) -> Self {
+        Self(value)
+    }
+
+    /// Creates a SignedDecimal256(Int256(value))
+    /// This is equivalent to `SignedDecimal256::from_atomics(value, 18)` but usable in a const context.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// # use cosmwasm_std::SignedDecimal256;
+    /// assert_eq!(SignedDecimal256::raw(1234i128).to_string(), "0.000000000000001234");
+    /// ```
+    pub const fn raw(value: i128) -> Self {
+        Self(Int256::from_i128(value))
+    }
+
+    /// Create a 1.0 SignedDecimal256
+    #[inline]
+    pub const fn one() -> Self {
+        Self(Self::DECIMAL_FRACTIONAL)
+    }
+
+    /// Create a -1.0 SignedDecimal256
+    #[inline]
+    pub fn negative_one() -> Self {
+        Self(-Self::DECIMAL_FRACTIONAL) // TODO: constify
```

Comment:
I guess, you wanted to define -1 constant for SignedDecimal256. Is it still `TODO` or we skip ti?